crates/bevy_ecs/src/query/state.rs

```diff
@@ -1162,12 +1162,19 @@ impl<Q: WorldQuery, F: ReadOnlyWorldQuery> QueryState<Q, F> {
     }
 }
 
-/// An error that occurs when retrieving a specific [`Entity`]'s query result.
+/// An error that occurs when retrieving a specific [`Entity`]'s query result from [`QueryState`].
 // TODO: return the type_name as part of this error
 #[derive(Debug, PartialEq, Eq, Clone, Copy)]
 pub enum QueryEntityError {
+    /// The given [`Entity`]'s components do not match the query.
+    /// 
+    /// Either it does not have a requested component, or it has a component which the query filters out.
     QueryDoesNotMatch(Entity),
+    /// The given [`Entity`] does not exist.
     NoSuchEntity(Entity),
+    /// The [`Entity`] was requested mutably more than once.
+    /// 
+    /// See [`QueryState::get_many_mut`] for an example.
```

Comment:
```suggestion
    /// See [`Query::get_many_mut`] for a workaround of this problem.
```

Reply:
I'd suggest linking to docs that the user is more likely to be familiar with.

Reply:
Again, I agree with you that people are more familiar with `Query` than `QueryState`, but there is the import thing, and also, I was referencing `QueryState::get_many_mut` because it has an example in the doc about how the error might get triggered, `Query::get_many_mut` does not and I don't understand how it would be "a workaround of this problem", it's using that method that can trigger the problem (among other things maybe?), and it's not documented there only in `QueryState`.

Reply:
Hmm. just checked the example, and I misunderstood the error. Whoops. For this one I'd just ignore this review comment.